cmd/notation/login.go

```diff
@@ -65,6 +68,23 @@ func runLogin(ctx context.Context, opts *loginOpts) error {
 	// initialize
 	serverAddress := opts.server
 
+	// input username and password by prompt
+	reader := bufio.NewReader(os.Stdin)
+	var err error
+	if opts.Username == "" {
+		opts.Username, err = readUsernameFromPrompt(reader)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.Password == "" {
+		opts.Password, err = readPasswordFromPrompt(reader)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Should this part be refactored with the `readPassword()` function in `PreRunE`?

Reply:
I think prompt happens in RunE instead of PreRunE. It is very different with readPassword() in preRunE which reads from piped input. Let me know if you have better ideas to refactor it.